File: Json/src/ec/edu/espe/json/controller/ProductController.java
package ec.edu.espe.json.controller;

import ec.edu.espe.json.model.Product;
import ec.edu.espe.json.model.ProductModel;

/**
 * Arelys Otavalo
 */
public class ProductController {

    private ProductModel model;

    public ProductController(ProductModel model) {
        this.model = model;
    }

    public void create(String name, float price) {
        model.addProduct(name, price);
    }

    public void list() {
        for (Product p : model.getAll()) {
            System.out.println(p);
        }
    }

    public Product search(int id) {
        return model.findById(id);
    }

    public boolean update(int id, String newName, float newPrice) {
        Product p = model.findById(id);
        if (p != null) {
            p.setName(newName);
            p.setPrice(newPrice);
            return true;
        }
        return false;
    }

    public boolean delete(int id) {
        return model.delete(id);
    }
}

File: EjemplosJson/JavaScript/CRUDJavascriptShowTotalInTerminalPRODUCT/Controller/ProductController.js
const ProductCrud = require("../Model/ProductCrud");

class ProductController {

    constructor() {
        this.crud = new ProductCrud();
    }

    add(product) {
        this.crud.addProduct(product);
    }

    getAll() {
        return this.crud.getAllProducts();
    }

    update(id, product) {
        return this.crud.updateProduct(id, product);
    }

    delete(id) {
        return this.crud.deleteProduct(id);
    }

    calculateTotal() {
        const products = this.crud.getAllProducts();
        let total = 0;

        for (let i = 0; i < products.length; i++) {
            total += products[i].price * products[i].quantity;
        }

        return total;
    }
}

module.exports = ProductController;


/*case "1":
    console.log("\n--- Product List ---");
    const products = controller.getAll();
    products.forEach(product => {
        const totalPrice = product.price * product.quantity;
        console.log(`ID: ${product.id}, Name: ${product.name}, Price: ${product.price}, Quantity: ${product.quantity}, Total Price: ${totalPrice}`);
    });
    menu();
    break;
    
    SHOW PRICE FOR EVERY PRUDUCT
*/ 